packages/search-in-workspace/src/node/ripgrep-search-in-workspace-server.ts

```diff
@@ -94,45 +95,101 @@ export class RipgrepSearchInWorkspaceServer implements SearchInWorkspaceServer {
+        const args = new Set<string>();
+
+        const appendGlobArgs = (rawPatterns: string[], exclude: boolean) => {
+            rawPatterns.forEach(rawPattern => {
+                if (rawPattern !== '') {
+                    const globArguments = this.patternToGlobCLIArguments(rawPattern, exclude);
+                    globArguments.forEach(arg => args.add(arg));
+                }
+            });
+        };
+
+        args.add('--hidden');
+        args.add('--json');
+
+        if (options && options.matchCase) {
+            args.add('--case-sensitive');
+        } else {
+            args.add('--ignore-case');
+        }
+
         if (options && options.includeIgnored) {
-            args.push('--no-ignore');
+            args.add('--no-ignore');
         }
         if (options && options.maxFileSize) {
-            args.push('--max-filesize=' + options.maxFileSize.trim());
+            args.add('--max-filesize=' + options.maxFileSize.trim());
         } else {
-            args.push('--max-filesize=20M');
+            args.add('--max-filesize=20M');
         }
+
         if (options && options.include) {
-            for (const include of options.include) {
-                if (include !== '') {
-                    args.push('--glob=**/' + include);
-                }
-            }
+            appendGlobArgs(options.include, false);
         }
+
         if (options && options.exclude) {
-            for (const exclude of options.exclude) {
-                if (exclude !== '') {
-                    args.push('--glob=!**/' + exclude);
-                }
-            }
+            appendGlobArgs(options.exclude, true);
         }
+
         if (options && options.useRegExp || options && options.matchWholeWord) {
-            args.push('--regexp');
+            args.add('--regexp');
         } else {
-            args.push('--fixed-strings');
-            args.push('--');
+            args.add('--fixed-strings');
+            args.add('--');
         }
-        return args;
+
+        return Array.from(args);
     }
 
-    // Search for the string WHAT in directories ROOTURIS.  Return the assigned search id.
+    /**
+     * Transforms a given file pattern to 'ripgrep' glob CLI arguments.
+     */
+    protected patternToGlobCLIArguments(pattern: string, exclude: boolean): string[] {
+        const toCLIGlobArgument = (inputPattern: string): string => {
+            const globCommandArgument = '--glob=';
+            const excludeChar = exclude ? '!' : '';
+            const subDirGlobPattern = '**/';
+
+            const subDirGlobPrefix = inputPattern.startsWith('/') ? '**' : subDirGlobPattern;
+            const updatedPattern = inputPattern.startsWith(subDirGlobPattern) ?
+                inputPattern : `${subDirGlobPrefix}${inputPattern}`;
+
+            return `${globCommandArgument}${excludeChar}${updatedPattern}`;
+        };
+
+        const globArgument = toCLIGlobArgument(pattern);
```

Comment:
Please inline `toCLIGlobArgument` here.

Reply:
I can not do it on line 161 above as the resulting globArgument is used in three other places below, 
so I would end up repeating the call... could you please clarify !

Reply:
```ts
    protected patternToGlobCLIArguments(pattern: string, exclude: boolean): string[] {
        const globCommandArgument = '--glob=';
        const excludeChar = exclude ? '!' : '';
        const subDirGlobPattern = '**/';

        const subDirGlobPrefix = inputPattern.startsWith('/') ? '**' : subDirGlobPattern;
        const updatedPattern = inputPattern.startsWith(subDirGlobPattern) ?
            inputPattern : `${subDirGlobPrefix}${inputPattern}`;

        const globArgument = `${globCommandArgument}${excludeChar}${updatedPattern}`;

        // ...
```

Reply:
@marechal-p , the intention of the function is to keep functions smaller to keep the SW readable.
see: https://refactoring.guru/smells/long-method

Having it nested may reduce a bit the sensation of a small function so I have removed it as suggested.
Something to think and discuss in general, so we can apply the best practices on future commits.

Reply:
To write it down here: Since the `toCLIGlobArgument` was local to the calling function and only ever used one it was useless.